core/lib/src/rocket.rs

```diff
@@ -445,14 +449,22 @@ impl Rocket {
                           Paint::default(LoggedValue(value)).bold());
         }
 
-        Rocket {
+        let (shutdown_sender, shutdown_receiver) = mpsc::channel(1);
+
+        let rocket = Rocket {
             config,
             router: Router::new(),
             default_catchers: catcher::defaults::get(),
             catchers: catcher::defaults::get(),
             state: Container::new(),
             fairings: Fairings::new(),
-        }
+            shutdown_handle: ShutdownHandle(shutdown_sender),
+            shutdown_receiver: Some(shutdown_receiver),
+        };
+
+        rocket.state.set(rocket.shutdown_handle.clone());
```

Comment:
Can we put a `// TODO` here regarding the type that will be put into managed state? I'm not sure we want to expose `State<ShutdownHandle>` as a way to get one.

Reply:
Resolving in favor of a newtype.